### PR TITLE
unnannounce device sync when sync view closes

### PR DESCRIPTION
**Pull request by karissa**
depends upon mapeo-server bump

### CODE REVIEW

File: src/components/Views/SyncView/SyncView.js [hunk 1]
```diff
@@ -84,10 +85,14 @@ class SyncView extends React.Component<
   }
 
   componentWillUnmount() {
+    const { unannounceSync } = nextProps;
+
     if (this.interval) {
       clearInterval(this.interval);
     }
 
+    unannounceSync();
```
Comment (noffle) [inline src/components/Views/SyncView/SyncView.js]: Is it guaranteed that `componentWillUnmount` will be called when the view closes?

**Comment by noffle** [comment]
Hey @czengg does this look good to you? I've already merged the server dependency for it at https://github.com/digidem/mapeo-server/pull/25.

File: src/components/Views/SyncView/SyncView.js (hunk 1 above)
Comment (czengg) [reply src/components/Views/SyncView/SyncView.js]: No, it is not guaranteed that the view will unmount. You might need to hook into the willBlur listener on react-navigation though: https://reactnavigation.org/docs/en/navigation-prop.html#addlistener-subscribe-to-updates-to-navigation-lifecycle

**Comment by czengg** [comment]
@noffle one quick comment otherwise looks good

File: src/components/Views/SyncView/SyncView.js (hunk 1 above)
Comment (noffle) [reply src/components/Views/SyncView/SyncView.js]: @czengg If we are then announcing on `componentDidMount` but unannouncing on `willBlur`, could we get into a state where we can "unblur" without re-mounting and have our announcing be turned off?

I'm feeling a little out of my depth here, and karissa is taking some time off. Looking to follow your lead on this. :)

**Comment by noffle** [comment]
@czengg I can't find your comment. Did you see mine on the unmount/blur stuff?

File: src/components/Views/SyncView/SyncView.js (hunk 1 above)
Comment (czengg) [reply src/components/Views/SyncView/SyncView.js]: @noffle Sorry just saw this. I can take over this branch if you'd like. We will need to check willFocus as well to trigger an announce. We shouldn't rely on the component lifecycle for this

**Comment by noffle** [comment]
@czengg yes please! I'm out of my depth here :)

**Comment by czengg** [comment]
@gmaclennan This branch is just about ready. It requires the update-observation branch to be merged however, because of the mapeo-server upgrade

**Pull request merged**